app/models/service_template_transformation_plan.rb

```diff
@@ -65,46 +66,23 @@ def self.create_catalog_item(options, _auth_user = nil)
     end
   end
 
-  private
+  def update_catalog_item(options, _auth_user = nil)
+    raise _("Editing a plan in progress is prohibited") if %w(active pending).include?(miq_requests.last.try(:request_state))
 
-  def enforce_single_service_parent(_resource)
-  end
-
-  def self.validate_config_info(options)
-    config_info = options[:config_info]
-
-    mapping = if config_info[:transformation_mapping_id]
-                TransformationMapping.find(config_info[:transformation_mapping_id])
-              else
-                config_info[:transformation_mapping]
-              end
-
-    raise _('Must provide an existing transformation mapping') if mapping.blank?
-
-    pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
-    post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
-
-    vms = []
-    if config_info[:actions]
-      vm_objects = VmOrTemplate.where(:id => config_info[:actions].collect { |vm_hash| vm_hash[:vm_id] }.compact).index_by(&:id).stringify_keys
-      config_info[:actions].each do |vm_hash|
-        vm_obj = vm_objects[vm_hash[:vm_id]] || vm_hash[:vm]
-        next if vm_obj.nil?
-
-        vm_options = {}
-        vm_options[:pre_ansible_playbook_service_template_id] = pre_service_id if vm_hash[:pre_service]
-        vm_options[:post_ansible_playbook_service_template_id] = post_service_id if vm_hash[:post_service]
-        vms << {:vm => vm_obj, :options => vm_options}
-      end
+    if miq_requests.any? || options[:config_info].nil?
+      update_attributes(:name => options[:name], :description => options[:description])
+      return reload
     end
 
-    raise _('Must select a list of valid vms') if vms.blank?
+    added_vms_enhanced_config_info = validate_config_info(options)
 
-    {
-      :transformation_mapping => mapping,
-      :vms                    => vms,
-      :provision              => config_info[:provision] || {}
-    }
+    transaction do
+      vm_resources.destroy_all
```

Comment:
Are we allowed to change the selection of infrastructure mapping? why not?

Reply:
We don't allow editing the associated infrastructure mapping.

I think the rationale here is that if one needs to change the infrastructure mapping for a plan, then creating a brand new plan with the desired infra mapping is a better approach.
Plus it's easier to deal with a static infrastructure mapping in the Wizard... as @mturley mentioned above https://github.com/ManageIQ/manageiq/pull/17989#discussion_r219550570, it complicates the state of the wizard.

Reply:
UI may forbid updating the infrastructure mapping, but what about API or a direct call for update that includes a changed mapping? We should either log a warning or raise an error if such change is detected.

Reply:
I re-read https://github.com/ManageIQ/manageiq/pull/17989#discussion_r219550570, and I think that we should consider supporting the infrastructure mapping update purely from a backend standpoint (API or a direct call, like you said)

UI may forbid updating it today, but that restriction may not apply in the future (based on the comment)

Reply:
added support for updating associated transformation mapping